### PR TITLE
Fix usage/price conversions between double and decimals

### PR DESCRIPTION
**Pull request by nieznanysprawiciel**
resolves: https://github.com/golemfactory/gamerhash-facade/issues/175

> [!IMPORTANT]  
> Changes are backward compatible, but GH app must update it's GolemLib dependency, to make the difference.

### CODE REVIEW

File: GolemLib/types.cs [hunk 1]
```diff
@@ -14,8 +15,14 @@ public class GolemUsage : GolemPrice
...
     // for compatibility with Rust ::std::f64::DIGITS this is hardcoded to 15
-    private const int DIGITS = 15;
-    private static decimal Round(decimal v) => Math.Round(v, DIGITS);
+    public static decimal Round(decimal v) => GolemUsage.RoundThroughDouble(v);
+
+    private static decimal RoundThroughDouble(decimal i)
+    {
+        // Print to string with exponential notation including 15 significant digits.
+        var formattedDouble = ((double)i).ToString("E15", CultureInfo.InvariantCulture);
+        return decimal.Parse(formattedDouble, System.Globalization.NumberStyles.Float);
```
[the remark this answers is not in the document]
Comment (prawilny) [reply GolemLib/types.cs:24]: Also please add the comment that we mean significant digits, not decimal numbers. It took me a minute to get why you didn't just use `Round()`.

[the remark this answers is not in the document]
Comment (prawilny) [reply GolemLib/types.cs:24]: Also I think that you misunderstood what Rust says in [the docs](https://doc.rust-lang.org/std/primitive.f64.html#associatedconstant.DIGITS):
"This is the maximum x such that **any** decimal number with x significant digits can be converted to f64 and back without loss."
The point is that Rust supports the very same IEEE754 floats which use 52 bits for significant digits - meaning that - just like in C# - **all** 15-base10-digits and **some** 17-base10-digits numbers are representable.

For example, see [the visualization](https://bartaz.github.io/ieee754-visualization/) and input [Rust `f64::MAX`](https://doc.rust-lang.org/std/primitive.f64.html#associatedconstant.MIN) there. Note that this number has 17 significant base10 digits. 

In other words: just convert decimal to IEEE 754 float whatever it is called in C#.

**Review by prawilny — CHANGES_REQUESTED** [review]

File: GolemLib/types.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (prawilny) [reply GolemLib/types.cs:24]: Also, please remove the misleading comment above.

**Review by prawilny — APPROVED** [review]
